medusa/post_processor.py

```diff
@@ -287,6 +288,16 @@ def list_associated_files(self, file_path, base_name_only=False, subtitles_only=
 
         return file_path_list
 
+    @staticmethod
+    def rar_basename(filepath, filelist):
+        videofile = os.path.basename(filepath)
+        rars = (x for x in filelist if rarfile.is_rarfile(x))
+
+        for rar in rars:
+            content = rarfile.RarFile(rar).namelist()
```

Comment:
Already explained: we can't and don't need to assume that the rar file in the folder is the one that contains the video file we are processing. We can make sure it is with this code. It has nearly no impact on resources and avoids incorrect associations.